androidApp/src/main/java/com/mbta/tid/mbta_app/android/util/NavHostControllerExtension.kt

```diff
@@ -0,0 +1,34 @@
+package com.mbta.tid.mbta_app.android.util
+
+import androidx.navigation.NavHostController
+import androidx.navigation.NavOptionsBuilder
+import com.mbta.tid.mbta_app.android.SheetRoutes
+
+/** If the current nav entry is StopDetails, this is the selected stop ID, otherwise null */
+val NavHostController.selectedStopId: String?
+    get() = this.currentRoute<SheetRoutes.StopDetails>()?.stopId
+
+/** Return the current stack entry as the provided type if they match, otherwise return null */
+inline fun <reified T : SheetRoutes> NavHostController.currentRoute(): T? =
+    this.currentBackStackEntry?.let { SheetRoutes.fromNavBackStackEntry(it) as? T }
+
+/** Return true if the current nav entry matches the provided type */
+inline fun <reified T : SheetRoutes> NavHostController.currentRouteMatches(): Boolean =
+    this.fromCurrentRoute { it?.let { it::class == T::class } } == true
+
+/**
+ * Convert the current nav stack entry into a SheetRoute, then run the provided lambda on it and
+ * return the result.
+ */
+fun <T> NavHostController.fromCurrentRoute(take: (SheetRoutes?) -> T): T =
+    this.currentBackStackEntry?.let { take(SheetRoutes.fromNavBackStackEntry(it)) } ?: take(null)
+
```

Comment:
nitpick(non-blocking): This taking a lambda to transform kind of makes it seem like this is doing two separate things where it might not need to. I think it might be clearer to chain the `let` / transformation after the call to convert into a SheetRoute outside this function.